STDEV-7843

apliter opened the pull request:
Добавлены новые команды для открытия чека коррекции под ФФД >= 1.1
Добавлены новые типы чеков и SettlementType.kt
Deprecated методы KktApi.kt

Commit:
STDEV-7843
Добавлены новые команды для открытия чека коррекции под ФФД >= 1.1
Добавлены новые типы чеков и SettlementType.kt
Deprecated методы KktApi.kt

Commit:
STDEV-7843
Изменена версия для публикации в локальном репозитории

Commit:
STDEV-7843
codestyle

xcam commented on src/main/java/ru/evotor/framework/core/action/command/open_receipt_command/OpenCorrectionIncomeReceiptCommand.kt:
```diff
@@ -0,0 +1,87 @@
+        }
+    }
+
+    fun getChanges(): List<PositionAdd> = this.changes
```
почему просто поля не сделать public?

apliter replied:
+

xcam commented on src/main/java/ru/evotor/framework/kkt/api/KktApi.kt:
```diff
@@ -222,6 +222,10 @@ object KktApi {
      */
+    @Deprecated(
+            message = "Неприменим для касс, использующих ФФД 1.1 и выше",
+            replaceWith = ReplaceWith("OpenCorrectionIncomeReceiptCommand \nOpenCorrectionOutcomeReceiptCommand")
```
Здесь вроде больше подходит PrintSellReceiptCommand

apliter replied:
Задача требует обычного цикла жизни чека в POS-е (со всеми event-ами). Поэтому использую именно Open* команды. 
Можно в догрузку еще и Print* команды сделать, или расширить имеющиеся Sell\Payback команды.

apliter replied:
+

xcam reviewed (changes requested):
см замечания

Commit:
STDEV-7843
resolve discussion #1

Commit:
Новая dokka

Commit:
STDEV-7843
up version

Commit:
STDEV-7843
up version

Commit:
STDEV-7843
little fix

Commit:
STDEV-7843
Добавлены броадкаст евенты для чеков коррекции.
Начата работа над командами PrintCorrectionIncomeReceiptCommand.kt и PrintCorrectionOutcomeReceiptCommand.kt

Commit:
STDEV-7843
Доделаны PrintCorrectionIncomeReceiptCommand.kt и PrintCorrectionOutcomeReceiptCommand.kt
Добавлены action-ы для событий чеков коррекции

Commit:
STDEV-7843
Добавлены поля для чеков коррекции в PrintCorrectionIncomeReceiptCommand.kt и PrintCorrectionOutcomeReceiptCommand.kt

Commit:
STDEV-7843
resolve discussion

Commit:
STDEV-7843
Добавлены новые методы в NavigationApi.kt

Commit:
Merge branches 'STDEV-7843' and 'develop' of https://github.com/evotor/integration-library into STDEV-7843

 Conflicts:
	build.gradle
	gradle.properties
	gradle/wrapper/gradle-wrapper.properties

xcam commented on src/main/java/ru/evotor/framework/core/action/command/open_receipt_command/OpenCorrectionIncomeReceiptCommand.kt:
```diff
@@ -0,0 +1,82 @@
+ * Команда открытия чека коррекции прихода
+ * @param changes Список позиций
+ * @param extra Дополнительные поля для чека
+ * @param correctionDate Дата совершения корректируемого расчета (ТЕГ 1178)
```
для ТЕГ 1178 у нас есть специальный класс: FiscalTag. Предлагаю использовать его

apliter replied:
+

xcam commented on src/main/java/ru/evotor/framework/core/action/command/open_receipt_command/OpenCorrectionIncomeReceiptCommand.kt:
```diff
@@ -0,0 +1,82 @@
+ * @param changes Список позиций
+ * @param extra Дополнительные поля для чека
+ * @param correctionDate Дата совершения корректируемого расчета (ТЕГ 1178)
+ * @param correctionType Тип коррекции BY_SELF - самостоятельная операция, BY_PRESCRIBED - операция по предписанию налогового органа  (ТЕГ 1173)
```
тут не надо расписывать возможные значения. Иначе при добавлении значения в enum 100% что сюда забудем добавить

apliter replied:
+

xcam commented on src/main/java/ru/evotor/framework/kkt/api/KktApi.kt:
```diff
@@ -222,6 +222,10 @@ object KktApi {
      * @param correctionDescription описание коррекции
      * @param callback
      */
+    @Deprecated(
```
я вот думаю не стоит ли под капотом в эвопосе сделать поддержку этого метода и для 1.1?
Мы в теории можем формировать чек коррекции с 1 позицией на всю сумму коррекции и 1 стандартной оплатой

apliter replied:
Да, сделано почти так. 
Методы kktApi будут вызывать print* команду.

xcam commented on src/main/java/ru/evotor/framework/core/action/command/print_receipt_command/PrintCorrectionOutcomeReceiptCommand.kt:
```diff
@@ -0,0 +1,131 @@
+ * @param correctionType Тип коррекции BY_SELF - самостоятельная операция, BY_PRESCRIBED - операция по предписанию налогового органа  (ТЕГ 1173)
+ * @param prescription Номер предписания налогового органа (ТЕГ 1179)
+ */
+class PrintCorrectionOutcomeReceiptCommand(
```
этот метод будет работать с ФФД 1.05?

apliter replied:
В теории - да

apliter replied:
Только если будет передана одна позиция. Больше одной - ошибка в callback-е

xcam reviewed (changes requested):
См комментарии

Commit:
STDEV-7843
resolve discussion

Commit:
STDEV-7843
Добавлены команды для чека коррекции возврата прихода и расхода

Commit:
Merge branch 'develop' of https://github.com/evotor/integration-library into STDEV-7843

Commit:
STDEV-7843
Добавлены экшны

Commit:
STDEV-7843
Добавлены рессиверы

apliter commented:
переезжаем в https://github.com/evotor/integration-library/pull/371
